### PR TITLE
Don't bundle libtpu in nightly

### PR DESCRIPTION
**Pull request by alanwaketan**
Summary:
Pallas depends on JAX to run, and JAX by default will use the system libtpu and won't use our own bundled libtpu. In order to have JAX and PyTorch/XLA run side by side, we will need to have them use the same libtpu. The solution could either be setting the TPU_LIBRARY_PATH or having both of them use the same system libtpu. The latter yields a much better UX.

Test Plan:
N.A.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -207,16 +207,16 @@ You can also add `+yyyymmdd` after `torch_xla-nightly` to get the nightly wheel
 of a specified date. To get the companion pytorch and torchvision nightly wheel,
 replace the `torch_xla` with `torch` or `torchvision` on above wheel links.
 
-#### Installing libtpu (before PyTorch/XLA 2.0)
+#### Installing libtpu
```
Comment (will-cromar) [inline README.md:210]: Can you just delete this section? There are more up to date instructions to install libtpu up top.

**Review by will-cromar — APPROVED** [review]
> Pallas depends on JAX to run, and JAX by default will use the system libtpu and won't use our own bundled libtpu. In order to have JAX and PyTorch/XLA run side by side, we will need to have them use the same libtpu. The solution could either be setting the TPU_LIBRARY_PATH or having both of them use the same system libtpu. The latter yields a much better UX.

It's worth noting this is a trade-off. To have release versions of JAX and PyTorch installed side-by-side, you must use the bundled libtpu in PyTorch/XLA, unless by some coincidence the two releases depend on the same `libtpu-nightly` version. This is why we use the bundled version in Colab/Kaggle.

Conversely to install JAX as a _dependency_ of PyTorch/XLA (rather than having both side-by-side), you have to use the exact nightly version we specify that matches the nightly version of libtpu we depend on.

Have you already tried overriding just JAX's libtpu path with `TPU_LIBRARY_PATH` before the import?

```
os.environ['TPU_LIBRARY_PATH'] = ...
import jax
```

Unbundling by default SGTM even if that works since we did this for release builds. This would just potentially make pallas possible in Colab/Kaggle. Either way, there's going to be some form of mess or hackery unless pallas can be separated out of JAX, or we can align with JAX on a common release schedule for libtpu.

**Comment by alanwaketan** [comment]
@will-cromar Thanks for approving the change, yea, this works:
```
os.environ['TPU_LIBRARY_PATH'] = ...
```

File: README.md (hunk 1 above)
Comment (alanwaketan) [reply README.md]: Done.

**Pull request merged**